src/components/MerchantsPage/ContributionBar.tsx

```diff
@@ -1,43 +1,116 @@
 import * as React from 'react';
-import styled from 'styled-components';
+import { useCallback, useEffect, useMemo, useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import styled from 'styled-components';
 
+// We assume in this code that all amounts are > 0.
 interface Props {
-  totalDonations: number;
-  totalGiftCards: number;
+  donationsRaised: number;
+  giftAMealAmountRaised: number;
+  giftCardAmountRaised: number;
 }
 
-const ContributionBar = ({ totalDonations, totalGiftCards }: Props) => {
+const percentage = (raised: number, total: number) => raised / total * 100;
+
+const ContributionBar = ({
+  donationsRaised,
+  giftAMealAmountRaised,
+  giftCardAmountRaised,
+}: Props) => {
   const { t } = useTranslation();
+  const [isSmallScreen, setIsSmallScreen] = useState(false);
 
-  const progressWidth = (raised: number, total: number) => {
-    if (raised < total) return (raised / total) * 100;
-    return 100;
-  };
+  const handleResize = useCallback(() => {
+    if (window.innerWidth < 641) {
+      setIsSmallScreen(true);
+    } else {
+      setIsSmallScreen(false);
+    }
+  }, [setIsSmallScreen]);
+
+  useEffect(() => {
+    handleResize();
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, [handleResize]);
+
+  const totalRaised = useMemo(
+    () => donationsRaised + giftAMealAmountRaised + giftCardAmountRaised,
```

Comment:
This looks good but I don't think we usually use this formatting--would you mind running `yarn format`? It'll match your code to our standard if you don't mind!

Reply:
oh good call, ran it on the 2 jsx files i touched